lib/text_parser.dart

```diff
@@ -958,6 +963,97 @@ class TextParser extends StatelessWidget {
       child: widget,
     );
   }
+
+  Widget _addInlineSpanToNode({required InlineSpan inlineSpan, required Widget nodeParsed}) {
+    if (nodeParsed is CleanRichText) {
+      return _addInlineSpanToCleanRichText(inlineSpan: inlineSpan, richText: nodeParsed);
+    } if (nodeParsed is Column && nodeParsed.children.isNotEmpty) {
+      return _addInlineSpanToColumn(inlineSpan: inlineSpan, column: nodeParsed);
+    } else {
+      return CleanRichText(
+        _optimizeTextspan(TextSpan(
+          children: [
+            WidgetSpan(child: nodeParsed),
+            inlineSpan
+          ]
+        )),
+        maxLines: maxLines
+      );
+    }
+  }
+
+  Widget _addInlineSpanToCleanRichText({required InlineSpan inlineSpan, required CleanRichText richText}) {
+    final childRichText = richText.child;
+    if (childRichText is TextSpan) {
+      return CleanRichText(
+        _optimizeTextspan(TextSpan(
+          text: childRichText.text,
+          children: [
+            if (childRichText.children?.isNotEmpty == true)
+              ...childRichText.children!,
+            inlineSpan
+          ]
+        )),
+        maxLines: richText.maxLines,
+        textAlign: richText.textAlign,
+      );
+    } if (childRichText is LinkTextSpan) {
```

Comment:
why is this if-block needed at all, why special-case `LinkTextSpan`?

Reply:
Oh, Srr, That was my mistake.